dot/state/grandpa.go

```diff
@@ -217,11 +218,11 @@ func (s *GrandpaState) GetSetIDByBlockNumber(num uint) (uint64, error) {
 
 		// if the given block number is greater or equal to the block number of the set ID change,
 		// return the current set ID
-		if num <= changeUpper && num > changeLower {
+		if blockNumber <= changeUpper && blockNumber > changeLower {
```

Comment:
Set id change happens in the last block of a given set.

Say current set ID is `s` and at block `b` set is getting updated to `s+1`. Block `b` is part of the set `s`. `GetSetIDChange(setId)` returns the block at which set id was set to `setId`. so block b is smaller than any block in the set `setId`. Thus the condition `blockNumber > changeLower`

Now for `GetSetIDChange(setId+1)` return the last block in the `setId` (call it `b'`). So, now if you pick any block in the `setId` it could be equal to or smaller than `b'`. Thus, `blockNumber <= changeUpper`